doc/OnlineDocs/contributed_packages/mindtpy.rst

```diff
@@ -71,67 +107,179 @@ MindtPy also supports setting options for mip solver and nlp solver.
                                      nlp_solver='ipopt',
                                      tee=True)
 
-There are three initialization strategies in MindtPy: rNLP, initial_binary, max_binary. In OA and GOA strategies, the default initialization strategy is rNLP. In ECP strategy, the default initialization strategy is max_binary.
+There are three initialization strategies in MindtPy: ``rNLP``, ``initial_binary``, ``max_binary``. In OA and GOA strategies, the default initialization strategy is ``rNLP``. In ECP strategy, the default initialization strategy is ``max_binary``.
 
-Single tree implementation
----------------------------------------------
-
-MindtPy also supports single tree implementation of Outer Approximation (OA) algorithm, which is known as LP/NLP algorithm originally described in [`Quesada & Grossmann`_].
-The LP/NLP algorithm in MindtPy is implemeted based on the LazyCallback function in commercial solvers.
+LP/NLP based branch and bound algorithm(Single-tree implementation)
+-------------------------------------------------------------------
 
-.. _Quesada & Grossmann: https://www.sciencedirect.com/science/article/abs/pii/0098135492800288
+MindtPy also supports single-tree implementation of Outer Approximation (OA) algorithm, which is known as LP/NLP based branch and bound algorithm originally described in [`Quesada & Grossmann 1992`_].
+The LP/NLP based branch and bound algorithm in MindtPy is implemeted based on the LazyConstraintCallback function in commercial solvers.
 
+.. _Quesada & Grossmann 1992: https://www.sciencedirect.com/science/article/abs/pii/0098135492800288
 
 .. note::
 
-   The single tree implementation currently only works with CPLEX.  To
-   use LazyCallback function of CPLEX from Pyomo, the `CPLEX Python
-   API`_ is required.  This means both IBM ILOG CPLEX Optimization
-   Studio and the CPLEX-Python modules should be installed on your
-   computer.
+   In Pyomo, `persistent solvers`_ are necessary to set or register callback functions. The single tree implementation currently only works with CPLEX and GUROBI, more exactly ``cplex_persistent`` and ``gurobi_persistent``. To use `LazyConstraintCallback`_ function of CPLEX from Pyomo, the `CPLEX Python API`_ is required.  This means both IBM ILOG CPLEX Optimization Studio and the CPLEX-Python modules should be installed on your computer. To use `cbLazy`_ function of GUROBI from pyomo, `gurobipy`_ is required. 
```

Comment:
In Pyomo, `persistent solvers`_ are necessary to set or register callback functions. The single tree implementation currently only works with CPLEX and GUROBI, more exactly ``cplex_persistent`` and ``gurobi_persistent``. To use the `LazyConstraintCallback`_ function of CPLEX from Pyomo, the `CPLEX Python API`_ is required. This means both IBM ILOG CPLEX Optimization Studio and the CPLEX-Python modules should be installed on your computer. To use the `cbLazy`_ function of GUROBI from pyomo, `gurobipy`_ is required.